torch/_inductor/utils.py

```diff
@@ -606,6 +611,11 @@ def has_incompatible_cudagraph_ops(gm):
     for node in gm.graph.nodes:
         if str(node.target) in forbidden_set:
             return True
+        if hasattr(node.target, "tags"):
+            if torch.Tag.dynamic_output_shape in node.target.tags:
+                return True
+            if torch.Tag.data_dependent_output in node.target.tags:
+                return True
```

Comment:
Some operators are only data dependent or dynamic in some scenarios, such as [aten.Index.Tensor](https://github.com/pytorch/pytorch/blob/main/torch/_subclasses/fake_tensor.py#L538-L545). 

Can we refine this skip ?

Reply:
This seems like a giant footgun. We should write decompositions for these operators so they are guaranteed data dependent or not.

Reply:
What is the concrete proposal here ? make two separate versions of aten.Index.Tensor ? And similarly, `repeat_interleave` ?

Reply:
Yes.

Reply:
It regresses a number of models which can actually cudagraph fine